ocaml/parsing/parser.mly

```diff
@@ -170,13 +170,21 @@ let mkexp_with_modes ?(ghost=false) loc modes exp =
       Exp.apply ~loc ext [Nolabel, exp]
 
 let mkpat_with_modes modes pat =
+  (* Modes doesn't belong to the pattern; in particular, it should not be
+     mentioned in a pattern error message. Therfore, it's not covered by the
+     pattern's location. We mark it ghost for location well-nestedness. *)
```

Comment:
I would remove the first two sentences of this comment and the below comment. They don't strike me as useful. The third sentence is most useful. (I meant mention that it's ppxlib's well-nestedness checks.)